packages/@aws-cdk/aws-stepfunctions-tasks/lib/ecs/run-task.ts

```diff
@@ -0,0 +1,376 @@
+import * as ec2 from '@aws-cdk/aws-ec2';
+import * as ecs from '@aws-cdk/aws-ecs';
+import * as iam from '@aws-cdk/aws-iam';
+import * as sfn from '@aws-cdk/aws-stepfunctions';
+import * as cdk from '@aws-cdk/core';
+import { ContainerOverride } from '..';
+import { integrationResourceArn, validatePatternSupported } from '../private/task-utils';
+
+/**
+ * Basic properties for ECS Tasks
+ */
+export interface EcsRunTaskProps extends sfn.TaskStateBaseProps {
+  /**
+   * The ECS cluster to run the task on
+   */
+  readonly cluster: ecs.ICluster;
+
+  /**
+   * [disable-awslint:ref-via-interface]
+   * Task Definition used for running tasks in the service.
+   *
+   * Note: this must be TaskDefinition, and not ITaskDefinition,
+   * as it requires properties that are not known for imported task definitions
+   */
+  readonly taskDefinition: ecs.TaskDefinition;
+
+  /**
+   * Container setting overrides
+   *
+   * Key is the name of the container to override, value is the
+   * values you want to override.
+   *
+   * @default - No overrides
+   */
+  readonly containerOverrides?: ContainerOverride[];
+
+  /**
+   * In what subnets to place the task's ENIs
+   *
+   * @default - Private subnet if assignPublicIp, public subnets otherwise
+   */
+  readonly subnets?: ec2.SubnetSelection;
+
+  /**
+   * Existing security group to use for the tasks
+   *
+   * @default - A new security group is created
+   */
+  readonly securityGroup?: ec2.ISecurityGroup;
+
+  /**
+   * Assign public IP addresses to each task
+   *
+   * @default false
+   */
+  readonly assignPublicIp?: boolean;
+
+  /**
+   * An Amazon ECS launch type determines the type of infrastructure on which your
+   * tasks and services are hosted.
+   *
+   * @see https://docs.aws.amazon.com/AmazonECS/latest/developerguide/launch_types.html
+   */
+  readonly launchTarget: IEcsLaunchTarget;
+}
+
+/**
+ * An Amazon ECS launch type determines the type of infrastructure on which your tasks and services are hosted.
+ * @see https://docs.aws.amazon.com/AmazonECS/latest/developerguide/launch_types.html
+ */
+export interface IEcsLaunchTarget {
+  /**
+   * called when the ECS launch target is configured on RunTask
+   */
+  bind(_task: EcsRunTask): EcsLaunchTargetConfig;
+}
+
+/**
+ * Configuration options for the ECS launch type
+ */
+export interface EcsLaunchTargetConfig {
+  /**
+   * Additional parameters to pass to the base task
+   *
+   * @default - No additional parameters passed
+   */
+  readonly parameters?: { [key: string]: any };
```

Comment:
Better to be more specific here - `PlatformTarget`, `PlacementConstraints`, etc.

So as to not collide with parameters that the `EcsRunTask` class may own and configure.

Reply:
I think as the consuming class of the bind, `EcsRunTask` is in control of collisions and how it wants to handle them. Either it would set it's properties and then spread the bind provided properties or the inverse of that depending on the desired behaviour. strongly typing this takes away some of the power of usage that seems pretty commonplace in our repo.

started to try this. got hit with `Method and non-static non-readonly property names must use camelCase: LaunchType` when I try to use PascalCase for property names.  Is this still something you want me to pursue?

Reply:
Yes, it'll need to be camel case. 

It's unlikely that we will have many more targets or that a customer will write their own, so it's fine to leave it as is.